crates/sui-json-rpc-types/src/sui_transaction.rs

```diff
@@ -1091,19 +1091,27 @@ impl SuiProgrammableTransactionBlock {
                         return result_types;
                     };
                     for (arg, type_) in c.arguments.iter().zip(types) {
-                        if let &Argument::Input(i) = arg {
-                            result_types[i as usize] = type_;
+                        match arg {
+                            &Argument::Input(i) if (i as usize) < inputs.len() => {
+                                result_types[i as usize] = type_;
+                            }
+                            _ => {}
```

Comment:
You could just use the `get` api here instead which would return a None if its out of bounds. probably saffer to do it that way too in case this code changes in the future